airflow/macros/__init__.py

```diff
@@ -86,3 +87,32 @@ def _integrate_plugins():
                     "import from 'airflow.macros.[plugin_module]' "
                     "instead. Support for direct imports will be dropped "
                     "entirely in Airflow 2.0.".format(i=macro_name))
+
+
+def s3_location(schema, table, is_dest=True):
+    """
+    Return a s3 location based on schema and table name if it is in production or local file location.
+
+    @param str schema: Database name
+    @param str table: Table name
+    @param bool is_dest: the table is a destination table(true) or intermediate table(false).
+    @return: a string which represents the s3 location of the table if it runs in production,
+             or location file location
+    """
+    env = os.environ.get('SERVICE_INSTANCE', 'development').upper()
+    if env.lower() == 'development':
+        return 'file:/tmp/hive-warehouse/' \
+               'PRODUCTION/{schema}/{table}'.format(schema=schema,
+                                                    table=table)
+    else:
+        if is_dest:
+            # Destination table in production
+            return 's3://lyftqubole-iad/qubole/t/stfihs/' \
```

Comment:
`stfihs` is just `shifts` backwards - it probably came from a table that ended in "shifts". Let's either make this part of the path a transformation of the name table or remove it entirely.

It's probably worth understanding why we had this logic of reversing the *end* of the table name in the first place.
My best guess: we're optimizing for how S3 stores these objects internally, and we didn't to overload a single prefix. If we had used the first 6 characters of the table name, we'd overload common table prefixes like `dimens` and `rides_`, but the last 6 characters is higher-entropy.

Reply:
What are we optimizing for here? Using a single namespace doesn't seem like the greatest optimization strategy.